modules/internal/DefaultRectangular.chpl

```diff
@@ -1036,13 +1036,24 @@ module DefaultRectangular {
     }
 
     proc dsiReallocate(d: domain) {
+      on this {
+      //
+      // If both d and dom are default rectangular, this is pretty
+      // easy...
+      //
       if (d._value.type == dom.type) {
-        on this {
         var copy = new DefaultRectangularArr(eltType=eltType, rank=rank,
-                                            idxType=idxType,
-                                            stridable=d._value.stridable,
-                                            dom=d._value);
-        for i in d((...dom.ranges)) do
+                                             idxType=idxType,
+                                             stridable=d._value.stridable,
+                                             dom=d._value);
+        //
+        // TODO: Making this for into a forall ought to accelerate
+        // dsiReallocate() calls, yet doing so breaks due to uint/int
+        // interaction issues today.  Deserves more of a look...
+        // Does our standalone parallel iterator not have the same
+        // type flexibility as the serial iterator?
+        //
+        for i in d[(...dom.ranges)] do
```

Comment:
Yeah, it might be nice to fix that long term, but right now the choice of whether to build up a standalone or just l/f happens at parse time based off if we think it's zippered.